sycl/source/detail/graph_impl.hpp

```diff
@@ -161,9 +156,10 @@ class node_impl {
   /// Construct a node from another node. This will perform a deep-copy of the
   /// command group object associated with this node.
   node_impl(node_impl &Other)
-      : MSuccessors(Other.MSuccessors), MPredecessors(Other.MPredecessors),
-        MCGType(Other.MCGType), MNodeType(Other.MNodeType),
-        MCommandGroup(Other.getCGCopy()), MSubGraphImpl(Other.MSubGraphImpl) {}
+      : enable_shared_from_this(Other), MSuccessors(Other.MSuccessors),
```

Comment:
Any chance this can be ` = default;` now?

Reply:
Good question, I tried it out and got a compilation error
```
sycl/source/detail/graph_impl.hpp:158:3: note: ‘sycl::_V1::ext::oneapi::experimental::detail::node_impl::node_impl(sycl::_V1::ext::oneapi::experimental::detail::node_impl&)’ is implicitly deleted because the default definition would be ill-formed:
```

we're also calling `Other.getCGCopy()` to initialize `MCommandGroup` which I don't think could be provided by the compiler.